uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Menu/CiviAdapter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+use CRM_HRCore_Menu_Item as MenuItem;
+
+class CRM_HRCore_Menu_CiviAdapter {
+
+  /**
+   * Builds the navigation menu tree and returns in a format expected
+   * by Civi in order to generate the navigation menu. It accepts a
+   * Menu Item instance
+   *
+   * @param MenuItem $menuItem
+   *
+   * @return array
+   *  Sample return format:
+   *  [
+   *    'attributes' =>
+   *      [
+   *        'label' => 'Components',
+   *         'url' => 'civicrm/bla',
+   *         'permission' => 'sample permission',
+   *         'operator' => 'AND',
+   *      ],
+   *   ],
+   */
+  public function getNavigationTree(MenuItem $menuItem) {
+    $navID = 1;
+    //The parentId is zero here to indicate that the top level navigation items
+    //has no parents. Also the navID is incremented by 1 for each menu item generated
+    //in the returning array, it starts at 1 and is passed by reference here.
+    return self::buildMenuTree($menuItem->getChildren(), 0, $navID);
```

Comment:
now that this is not a static method anymore, we need to use `$this` instead of `self`